### ISSUE TITLE
Playing randomly stops between songs while screen is locked

### ISSUE THREAD

**Issue by LostCross** [kabouzeid/phonograph-issue-tracker#103]
When a song ends, sometimes the next one won't play until I open the screen. Then as soon as I do that, it resumes playing

**Comment by pabsrocka** [comment]
Here the same, i have to turn on the screen to start playing next song.
Try enabling gapless playback (yes, the setting says that this option may cause problems, but in our case, this option solves that, at least for me)

**Comment by moneytoo** [comment]
Same issue here with latest stock firmware on Lenovo Vibe Z2 Pro (K920). I have more or less the same apps on Samsung Galaxy S5 (CM12.1) but I it never occured on this phone while it's happening almost always with Lenovo.
Is there even any wakelock held between switching tracks?

**Comment by kabouzeid** [comment]
Yes there is a wakelock between switching songs.  We had that issue before, but I fixed it a long time ago, so I'm wondering why it appears again...

**Comment by gazzawazza** [comment]
Hi Karim

Just emailed you about this then thought to look on github (doh).

I've got the same problem. Music seems to stop playing (app doesn't crash though) after a while with the screen off. Turning screen on resumes playback. Think music stops at the end of a track.

Phonograph 0.9.5.2 build on sony z3 compact (android 5.1.1).

I've tried the suggested gapless playback workround.

1) the gapless playback works (doesn't seem to cause an issue on my handset)

2) It seems to help with the above described issue (#103) but I did witness one moment when the music sounded like it stopped for 10 seconds (between tracks) but then started again. I checked the two tracks and they did not have silence at the end or the beginning.

Very odd.


Regards,

Gary

**Comment by moneytoo** [comment]
After switching from CyanogenMod back to stock firmware on my Samsung Galaxy S5, I'm facing the issue here as well.

**Comment by gazzawazza** [comment]
I was a little surprised at what seemed to be quite a lot of battery usage too, while I was testing.

So, I compared performance to my old music app (Sony Music). 

I compared approx 1h 30m of (screen off) speaker playback using Sony music (91.3.A.0.0) and Phonograph (0.9.5.2).

**Phonograph**
CPU total 5m25s
CPU foreground 1m5s
keep awake 1h 29m 55s

**Sony Music**
CPU total 3m26s
CPU foreground 13s
keep awake 1h 30m 8s

Could the increased CPU total for photograph be because of the gapless playback?


Regards,

Gary
